txscript/reference_test.go

```diff
@@ -293,15 +345,36 @@ func testScripts(t *testing.T, tests [][]interface{}, useSigCache bool) {
 			continue
 		}
 
+		var (
+			witness  wire.TxWitness
+			inputAmt btcutil.Amount
+		)
+
 		// When the first field of the test data is a slice it contains
 		// witness data and everything else is offset by 1 as a result.
 		witnessOffset := 0
-		witnessData, ok := test[0].([]interface{})
-		if ok {
-			witnessOffset++
+		if witnessData, ok := test[0].([]interface{}); ok {
+
+			// If this is a witness test, then the final element
+			// within the slice is the input amount, so we ignore
+			// all but the last element in order to parse the
+			// witness stack.
+			strWitnesses := witnessData[:len(witnessData)-1]
+			witness, err = parseWitnessStack(strWitnesses)
+			if err != nil {
+				t.Errorf("%s: can't parse witness; %v", name, err)
+				continue
+			}
 
+			inputAmt, err = btcutil.NewAmount(witnessData[len(witnessData)-1].(float64))
+			if err != nil {
+				t.Errorf("%s: can't parse input amt: %v",
+					name, err)
+				continue
+			}
+
+			witnessOffset++
```

Comment:
Might want to move this to the top where the comment about it is, or move the comment down here and adjust the comment on the if statement accordingly.  I would prefer the latter personally.